### ISSUE TITLE
Error "Unhandled Rejection (NotSupportedError): Failed to set the 'buffer' property on 'ConvolverNode': The buffer sample rate of 48000 does not match the context rate of 44100 Hz."

### ISSUE THREAD

**Issue by micbuffa** [Kutalia/react-webaudio-5150#3]
Hi! When I compile and run the app on my computer (Mac Book Pro 2021), Catalina, I got a "Unhandled Rejection (NotSupportedError): Failed to set the 'buffer' property on 'ConvolverNode': The buffer sample rate of 48000 does not match the context rate of 44100 Hz."

 103 |         audioData[i] = impulseBufferArr[i];
  104 |     }
  105 | 
> 106 |     preampConvolver.buffer = audioBuffer;
      | ^  107 | 
  108 |     onPluginReady([preampConvolver, node], index);
  109 | });

**Comment by micbuffa** [comment]
Managed to run it setting up the AudioContext to 48Khz in App.tsx, but I don't understand why... your impulse files are at 44.1 Khz... ?

`const audioContext = state.audioContext || new AudioContext({
      latencyHint: 'interactive',
      sampleRate: 48000,
    });`

**Comment by Kutalia** [comment]
Hello. Thank you for the bug discovery and the fix. I have just pushed the commit.
The preamp IRs extracted from kpp_tubeamp .tapf profile files are in 48KHz format while custom cabinet IR files are in 44.1KHz. Seems like your machine is using 44.1KHz sampling rate by default so it conflicted with the preamp IR (48KHz).
The cabinet .wav IRs are resampled using `decodeAudioData` function so it has no problem working with any audioContext `sampleRate` setting, while the preamp IR sample rate is locked to 48KHz, therefore the `audioContext` needs to be in sync. Later I might add resampling algorithm when reading from those *.tapf files.

**Comment by micbuffa** [comment]
I understand… btw I tried the app and it sounds great. I have the same DI sample for guitar in my own amp sims :-) So I can compare… would you like to make a small visio one of these days? My email micbuffa@gmail.com. I’d like to talk about tube amp simulation, faust, plugins, fx for guitar etc. I made tWebAudio plugins out of Oleg’s work during the last year, and they are included in our pedalboard web app, but I never completed this work with his amp sim… glad to be able to try it with your app…  I wrote three amp sims: one for classic rock, one for metal and one for clean/acoustic sounds.And we created an « amp designer web app » so we can also generate different amp sims by choosing the number and type of tubes, filters, power amp settings, etc. I’m also part of a group of developers who created WebAudioModules V2, a standard for creating webaudio plugins, in JS, TS, react, Faust, sound, C++, etc. I’d like to demo that to you… Michel

**Comment by Kutalia** [comment]
I have overviewed your works and they've been extremely helpful for my project! I also got to know the first version of WebAudioModules which is few years old so I started this project from a scratch. Nice to see the second iteration which hopefully is more modern, polished and sustainable.
Responding to [this thread](https://github.com/Kutalia/react-webaudio-5150/issues/2) as well, I'll be watching the presentation of WebAudioModules V2 and contacting you through email.
